guides/source/generators.md

```diff
@@ -475,35 +496,255 @@ after_bundle do
 end
 ```
 
-First, the template asks the user whether they would like to install Devise.
-If the user replies "yes" (or "y"), the template adds Devise to the `Gemfile`,
-and asks the user for the name of the Devise user model (defaulting to `User`).
-Later, after `bundle install` has been run, the template will run the Devise
-generators and `bin/rails db:migrate` if a Devise model was specified. Finally, the
-template will `git add` and `git commit` the entire app directory.
+The above template adds Devise to the `Gemfile` after asking the user and also
+allows user to name the Device user model. After `bundle install` has been run,
+the template runs the Devise generators and `bin/rails db:migrate`. Finally, the
+template does `git add` and `git commit`.
 
-We can run our template when generating a new Rails application by passing the
-`-m` option to the `rails new` command:
+The above `template.rb` uses helper methods such as `after_bundle` and `rails_command`. You can see a list of available helper methods that templates can use in the [generator helper methods](#generator-helper-methods) section. The following sections explains some of them with examples.
 
-```bash
-$ rails new my_cool_app -m path/to/template.rb
+Rails Template API
+------------------
+
+The Rails template ruby scripts have access to several helper methods using a DSL (Domain Specific Language).
+
+Here's an example of a typical Rails template:
```

Comment:
Should we describe what the Rails template does?

Here's an example of a typical Rails template that ....